CICE-Consortium/CICE
# Add new unit tests sumchk and bcstchk and update tests

## PR checklist
- [X] Short (1 sentence) summary of your PR: 
    Update testing and add sumchk and bcstchk unit tests
- [X] Developer(s): 
    apcraig
- [X] Suggest PR reviewers from list in the column to the right.
- [X] Please copy the PR test results link or provide a summary of testing completed below.
    Full tests suites run on cheyenne on 3 compilers, https://github.com/CICE-Consortium/Test-Results/wiki/cice_by_hash_forks#d7d59a49665d432dfd06216d16915c3b52d4b98f.  Note that test results for tx1 change because the ns_boundary_type was changed from 'open' to 'tripole'.
- How much do the PR code changes differ from the unmodified code? 
    - [X] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
- Does this PR add any new test cases?
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [ ] No, does the documentation need to be updated at a later time?
- [X] Please provide any additional information or relevant details below:

Update testing
- Set default debug_model_step=0 (was 99999999)
- Add debug_model_[i,j,iblk,task] to define the debug_model diagnostic point in local grid index space.  If this point is not set and debug_model is turned on, it will use lonpnt(1),latpnt(1).
- Rename forcing_diag namelist/variable to debug_forcing to be more consistent with other "debug_" namelist variables
- Rename the local variable forcing_debug in ice_forcing.F90 to local_debug to avoid confusion with global varaible debug_forcing.
- Add namelist variable optics_file and optics_file_fieldname.  Was hardwired in ice_forcing_bgc.F90.
- Update setting of nbtrcr_sw and allocation of trcrn_sw.   nbtrcr_sw was not set in icepack after it was computed and trcrn_sw was allocated before nbtrcr_sw was computed.  This impacts the dedd_algae implementation which still isn't working.
- move default distribution_wgt_file for gx1 to set_nml.gx1
- update test suite, add testing of debug_model_[i,j,iblk,task], add addtional testing of maskhalo
- add sumchk unit test to test ice_global_reduction methods including on center and Nface locations
- add bcstchk unit test to test ice_broadcast methods
- add get_rank to ice_communicate.F90
- add global_[min/max]val_scalar_int_nodist method to ice_global_reductions.F90
- add tripole output in ice_blocks.F90 with debug_blocks
- add grid_type and ns_boundary_type tripole check
- update set_nml.tx1 to set ns_boundary_type to 'tripole', was 'open'.  **NOTE: this changes answers for tx1 tests**
- update lsum16 global sum to revert to double precision if NO_R16 is set
- update documentation
